File: demo_apps/vulnerable_bank_app/payment.js
const axios = require("axios");
const lodash = require("lodash");
const express = require("express");

const router = express.Router();

// Missing helmet setup — triggers detection
const app = express();

// Vulnerable: CORS wildcard
app.use(cors({ origin: "*" }));

// Vulnerable: TLS disabled
const paymentClient = axios.create({
  baseURL: "https://payment-gateway.bank.com",
  rejectUnauthorized: false,
});

// Vulnerable: prototype pollution
function processPayment(req) {
  const payload = lodash.merge({}, req.body);
  return paymentClient.post("/transfer", payload);
}

router.post("/pay", (req, res) => {
  processPayment(req)
    .then((result) => res.json(result.data))
    .catch((err) => res.status(500).json({ error: err.message }));
});

module.exports = router;

File: demo_apps/vulnerable_bank_app/payment.py
import requests
from flask import Flask, request, jsonify
from flask_cors import CORS

app = Flask(__name__)

# Vulnerable: CORS wildcard — mirrors app.use(cors({ origin: "*" }))
CORS(app, origins="*")

PAYMENT_GATEWAY = "BANK_URL"

# Vulnerable: TLS disabled — mirrors rejectUnauthorized: false
def process_payment(payload):
    return requests.post(
        f"{PAYMENT_GATEWAY}/transfer",
        json=payload,
        verify=False
    )

@app.route("/pay", methods=["POST"])
def pay():
    try:
        payload = request.json
        result = process_payment(payload)
        return jsonify(result.json())
    except Exception as e:
        return jsonify({"error": str(e)}), 500
